vent/sqldb/sqldb.go

```diff
@@ -1,338 +1,66 @@
 package sqldb
 
 import (
-	"database/sql"
-	"fmt"
+	"errors"
 
-	"github.com/lib/pq"
 	"github.com/monax/bosmarmot/vent/logger"
+	"github.com/monax/bosmarmot/vent/sqldb/adapters"
 	"github.com/monax/bosmarmot/vent/types"
 )
 
 // SQLDB implements the access to a sql database
 type SQLDB struct {
-	DB     *sql.DB
-	Log    *logger.Logger
-	Schema string
+	DBAdapter adapters.DBAdapter
 }
 
 // NewSQLDB connects to a SQL database and creates default schema and _bosmarmot_log if missing
-func NewSQLDB(dbURL string, schema string, l *logger.Logger) (*SQLDB, error) {
-	var found bool
```

Comment:
change comment
`// NewSQLDB connects to a SQL database and creates default schema and _bosmarmot_log if missing` to
`// NewSQLDB constructs an adapter for the given dbms`